### PR TITLE
handle queuing and popping from the queue

### PR DESCRIPTION
**Pull request by fishnix**
early implementation of queueing jobs and popping from the queue.

### CODE REVIEW

**Comment by damnski** [comment]
:+1:

File: jobs/queue.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+	"github.com/go-redis/redis"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type Queue struct {
+	Name       string
+	BackupName string
+	Window     int64
```
Comment (tenyo) [inline jobs/queue.go]: This is in seconds?

File: common/config_test.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ var testConfig = []byte(
 		},
 		"jobsRepository": {
 			"type": "s3",
-			"refreshPeriod": "60m",
+			"RefreshInterval": "60m",
```
[the remark this answers is not in the document]
Comment (fishnix) [reply common/config_test.go]: this will actually work as well, but i can update it to be consistent

File: jobs/queue.go (hunk 1 above)
Comment (fishnix) [reply jobs/queue.go]: yeah it was a copy pasta - i might rework some of that but for now i think its okay

**Pull request merged**